### PR TITLE
Swap out syscall for sys/unix

### PR DESCRIPTION
**Pull request by peterguy**
`syscall` has been frozen since Go 1.3 and deprecated
(https://go.dev/doc/go1.4#major_library_changes).

Using the replacement `golang.org/x/sys/unix` package will bring in bug fixes
and enhancements since `syscall` was frozen in 1.3,
and will pave the way for multi-platform builds, specifically for the native/single-binary project currently.

Also add a flag to a git call used by a test file so that it works with git 2.38.1+

### CODE REVIEW

File: shards/shards.go
Comment (keegancsmith) [inline shards/shards.go]: this is unrelated to the PR and breaks a fix that has landed. Similarly with the change to watcher.go

**Review by keegancsmith — APPROVED** [review]
Approving  to unblock, but it includes a change that should not land (mentioned inline). Thanks!

File: shards/shards.go
Comment (peterguy) [reply shards/shards.go]: I removed those changes - I had brought them in with a rebase or merge from main - and I rebased with changes that have been merged with main since this branch was created. Of note, I merged [#516](https://github.com/sourcegraph/zoekt/pull/516) - there were some conflicts to resolve.

**Comment by peterguy** [comment]
Alright @keegancsmith , I rebased to drop those other commits from main, then resolved the conflict between this PR and #516. Should be good to go now.

**Review by keegancsmith — APPROVED** [review]

**Comment by peterguy** [comment]
What's the next step now? I don't have permission to merge; will a merge be scheduled? Do I need to do anything more?

**Comment by keegancsmith** [comment]
Sorry, thought everyone in sourcegraph eng could merge. I'll do it for you and check out why you couldn't merge.

**Comment by keegancsmith** [comment]
Just checked, the specia lsourcegraph everyone group has write access to this repo and you are in it. In your next PR to zoekt if you still can't merge I'll dive deeper.

**Comment by peterguy** [comment]
ok, thanks!

**Pull request merged**
